librecad: fix missing app name and icon on Wayland

lightbulbjim opened the pull request:
###### Motivation for this change

Wayland uses the .desktop file to find the app name and icon. Without it being specified the correct icon is not shown. This was broken in the recent update to Qt5 (#69077).

Tested on Gnome/Wayland.

References:

  - https://community.kde.org/Guidelines_and_HOWTOs/Wayland_Porting_Notes#Application_Icon
  - https://doc.qt.io/qt-5/qguiapplication.html#desktopFileName-prop

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @viric

bjornfor commented:
I see you've submitted this upstream: https://github.com/LibreCAD/LibreCAD/pull/1151.

Perhaps we can apply some pressure to fix this upstream?

lightbulbjim commented:
I’ll do what I can. 

Since the next release is at RC status this change mightn't ship anytime soon, so it might be worth including the patch in nixpkgs for now.

> On 29 Oct 2019, at 02:56, Bjørn Forsman <notifications@github.com> wrote:
> 
> I see you've submitted this upstream: LibreCAD/LibreCAD#1151 <https://github.com/LibreCAD/LibreCAD/pull/1151>.

> Perhaps we can apply some pressure to fix this upstream?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/NixOS/nixpkgs/pull/71957?email_source=notifications&email_token=ABA42VHH2TP445K72BKSZ43QQ4DTVA5CNFSM4JE5L3G2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECNMPUY#issuecomment-547014611>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABA42VDOTQ324C4QX775Q5TQQ4DTVANCNFSM4JE5L3GQ>.

lightbulbjim committed:
librecad: fix missing app name and icon on Wayland

Wayland uses the .desktop file to find the app name and icon. Without it being specified the correct icon is not shown.

Tested on Gnome/Wayland.

References:

  - https://community.kde.org/Guidelines_and_HOWTOs/Wayland_Porting_Notes#Application_Icon
  - https://doc.qt.io/qt-5/qguiapplication.html#desktopFileName-prop

Kiwi reviewed (changes requested):
I'd like if you run nixpkgs-fmt on it; alternatively `git apply` :)
```
diff --git a/pkgs/applications/misc/librecad/default.nix b/pkgs/applications/misc/librecad/default.nix
index 2d160d93853..6f7967dfdfd 100644
--- a/pkgs/applications/misc/librecad/default.nix
+++ b/pkgs/applications/misc/librecad/default.nix
@@ -17,7 +17,7 @@ let
   stdenv = gcc8Stdenv;
 in

-# Doesn't build with gcc9
+  # Doesn't build with gcc9
 mkDerivationWith stdenv.mkDerivation rec {
   pname = "librecad";
   version = "2.2.0-rc1";
@@ -31,11 +31,13 @@ mkDerivationWith stdenv.mkDerivation rec {

   patches = [
     ./fix_qt_5_11_build.patch
-    (fetchpatch {
-      # Fix missing app name and icon on Wayland.
-      url = "https://github.com/LibreCAD/LibreCAD/commit/a17f8281093403f0c7c36996232665ed21906688.patch";
-      sha256 = "1x46psh4bcx2hxck4l83ki43g1252vb033i2x94h4rpai9hww4d5";
-    })
+    (
+      fetchpatch {
+        # Fix missing app name and icon on Wayland.
+        url = "https://github.com/LibreCAD/LibreCAD/commit/a17f8281093403f0c7c36996232665ed21906688.patch";
+        sha256 = "1x46psh4bcx2hxck4l83ki43g1252vb033i2x94h4rpai9hww4d5";
+      }
+    )
   ];

   postPatch = ''
```

lightbulbjim commented:
> I'd like if you run nixpkgs-fmt on it; alternatively `git apply` :)

Sure thing, done. That comment indent looks awkward to me, but no big deal.

Kiwi commented:
Aye. I don't always agree with The Machine but when we disagree it always wins the argument! I'm a huge fan of code formatters. They're much more consistent than I am, with a lot less mental overhead, and it's nice to be able to defer the decision to The Machine about what is correct when it comes to something so subjective (and I don't even always agree with myself!). It helps me stay focused on the problem I'm solving if I don't have to think about how it's formatted as well. If it compiles... format it! I should write about it on my blag... anyway, thanks! :smile:

doronbehar commented:
Result of `nixpkgs-review pr 71957` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>librecad</li>
  </ul>
</details>
